app-marketplace-v2-spec/introduction.md

```diff
@@ -0,0 +1,96 @@
+# Introduction
+
+  * [Problem Statement](#problem-statement)
+  * [Project Summary](#project-summary)
+  * [Objectives](#objectives)
+    + [Full Backwards Compatibility](#full-backwards-compatibility)
+    + [Supported Use Cases](#supported-use-cases)
+    + [Launch App Marketplace](#launch-app-marketplace)
+    + [Scoped **Cloud App** API access](#scoped---cloud-app---api-access)
+    + [Meet **Mattermost Cloud** Operational
+      Requirements](#meet---mattermost-cloud---operational-requirements)
+  * [Technical Principles](#technical-principles)
+
+## Problem Statement
+
+The current extensibility framework supports http-based **Integrations** and
+Go/React **Plugins**.
+
+1. **Integrations** are based on simple HTTP1.1 messages, and provide very basic
+   interactive functionality, and are not easily hostable on-prem. "As is" this
+   framework is not sufficient to develop most interactive in-place use-cases.
+   They are mostly supported on mobile.
+2. **Plugins** do not offer significant capabilities on mobile. Historically,
+   they have been implemented with WebApp first in mind.
+3. **Plugins** require development in Go, React; expensive and slow to develop
+   and review.
+4. **Plugin** installation and configuration management UX is neither
+   consolidated, nor consistent.
+5. **Plugins** are not sufficiently isolated neither in terms of data access,
+   nor in terms of performance for cloud (and some enterprise) deployments.
+6. **Slack** compatibility has not been actively maintained, is behind.
+
+## Project Summary 
+
+1. Build the **App Marketplace**, the place to discover,
```

Comment:
If App Marketplace is going to be the place to manage plugins... does it need also capabilities to install and configure custom plugins? (i.e. plugins manually uploaded to the instance).

Reply:
0/5 - not yet. I propose that we keep all existing mechanisms in place, deprecate/consolidate them as a next "major release".